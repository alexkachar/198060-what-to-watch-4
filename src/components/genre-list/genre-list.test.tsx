import * as React from 'react';
import * as renderer from 'react-test-renderer';
import GenreList from './genre-list';
import MOCK_GENRES from '../../test-data/mock-genres';

const SELECTED_GENRE = `Action`;

it(`GenreList renders correctly`, () => {
  const tree = renderer
    .create(
        <GenreList
          genres={MOCK_GENRES}
          selectedGenre={SELECTED_GENRE}
          onGenreSelect={jest.fn()}
        />, {
          createNodeMock: () => document.createElement(`div`)
        })
    .toJSON();
  expect(tree).toMatchSnapshot();
});
