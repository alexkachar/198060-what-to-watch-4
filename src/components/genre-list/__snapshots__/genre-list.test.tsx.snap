// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Action
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Adventure
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Comedy
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Crime
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Drama
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Fantasy
    </a>
  </li>
  <li
    className="catalog__genres-item null"
  >
    <a
      className="catalog__genres-link"
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      Thriller
    </a>
  </li>
</ul>
`;
